Optimize, Correct, and Improve LenChk

terminalforlife opened the pull request:
Please refer to commit message for all the information on the changes. Let me know if I missed something.

chubin commented on tests/lenchk:
```diff
@@ -95,86 +97,83 @@ while [ "$1" ]; do
 					IFS=',' read -a Fields <<< "$CheatLine"
 					for Field in "${Fields[@]}"; {
-						[ "$Field" == "$Progrm=disable" ] && SkipFile='True'
+						[[ $Field == $Progrm=disable ]] && SkipFile='True'
```
Strictly speaking you need quotes around `$Program=disable`, in the RHS part (but not around `$Field`!)

terminalforlife replied:
Agreed. I wrote LenChk back before I had a brainwave moment, finally realising inefficient and unnecessary `[` is in BASH, these days.

terminalforlife replied:
Hi. Finally getting around to this, having crawled back from my year-long hiatus.

Although word-splitting doesn't apply with variables in either side of `[[`, it _is_ possible for someone to have glob pattern matching syntax in their path which leads up to the executable itself, which _would_ be interpreted as such. However, `$Progrm` is `${0##*/}`, leaving just the script name. Just in-case, because you never know what a user will do (lol), I will quote-protected it.

chubin commented on tests/lenchk:
```diff
@@ -95,86 +97,83 @@ while [ "$1" ]; do
 		# If the current file matches one which is whitelisted, skip it.
 		for CurWL in "${Whitelisted[@]}"; {
-			[ "$File" == "$CurWL" ] && continue 2
+			[[ $File == $CurWL ]] && continue 2
```
And here around `$CurWL` too (but not around `$File` !)

terminalforlife replied:
I will do the same here for the same reason as mentioned above.

chubin commented on tests/lenchk:
```diff
@@ -184,26 +183,26 @@ Main(){
 					# The line number of the problematic length.
-					[ "$NoColor" == 'True' ] || printf "$C_LineNums"
+					[[ $NoColor == True ]] || printf "$C_LineNums"
 					printf '  %7d  ' $LineNum # <-- allows for 9,999,999 lines.
```
Missing quotes around `$LineNum`?

chubin reviewed (commented):
I would suggest fix the highlighted (very minor) problems before the merge,
just to try to keep it as clean as possible. What do you think?

terminalforlife commented:
I'll go over this PR again — bear with me. Hope you're well, BTW. :)

chubin commented:
Hi @terminalforlife 
Are we going to merge this one?
I believe, it is almost done?

terminalforlife commented:
Yeah, sorry for the lack of input. Don't worry about merging it, at least for now. I'll revisit this in the future. LenChk should be fine as-is, for the time being. Feel free to close the PR, if you're just trying to clean up the repository. I still have the changes locally for a future PR.

sledzikteo commented:
ok

terminalforlife commented:
I can only apologise for what was a crappy PR. Although I knew the original iteration of LenChk worked, I still left this here unfinished for a _long_ time. I was burned out with Linux and programming, which ultimately led to a much needed break from it all.

I've addressed the issues and brought it up to my current standards. I have tested it, but please do ensure you're happy with it first by testing it on your end as well.
